### PR TITLE
Fix resetting of TTL

### PR DESCRIPTION
**Pull request by mattsaxon**
Previously the reset_ttl method changed the time created and the TTL value, but did not change the expiration time or stale times. As a result a record would expire even when this method had been called.

### CODE REVIEW

File: zeroconf/__init__.py [hunk 1]
```diff
@@ -503,6 +503,10 @@ def reset_ttl(self, other: 'DNSRecord') -> None:
         self.created = other.created
         self.ttl = other.ttl
 
+        """ reset expiration times """
```
Comment (jstasiak) [inline zeroconf/__init__.py]: This comment seems redundant, the code already expresses that.

Comment (mattsaxon) [reply zeroconf/__init__.py]: Agreed, removed in latest commit

**Comment by mattsaxon** [comment]
@jstasiak the failing test here look to me to be a problematic test, it appears to randomly pass/fail and is not related to this change.

**Comment by mattsaxon** [comment]
> Please provide a test for this.

Added as requested

File: zeroconf/test.py [hunk 2]
```diff
@@ -16,8 +17,11 @@
 
 from nose.plugins.attrib import attr
 
-import zeroconf as r
-from zeroconf import (
+# ensure I can find this package even when it hasn't been installed (for development purposes)
+sys.path.insert(0, '..')
```
Comment (jstasiak) [inline zeroconf/test.py]: This is better done by setting `PYTHONPATH` when running tests/importing. Manipulating `sys.path` (particularly at import time) is a dangerous affair and let's avoid doing it.

File: zeroconf/test.py [hunk 3]
```diff
@@ -16,8 +17,11 @@
...
+
+import zeroconf as r  # noqa: E402
+from zeroconf import (  # noqa: E402
```
Comment (jstasiak) [inline zeroconf/test.py]: Those `# noqa` pragmas are out of scope of this pull request.

File: zeroconf/test.py [hunk 4]
```diff
@@ -79,6 +83,22 @@ def test_dns_record_abc(self):
...
         self.assertRaises(r.AbstractMethodException, record.write, None)
 
+    def test_dns_record_reset_ttl(self):
+        record = r.DNSRecord('irrelevant', r._TYPE_SRV, r._CLASS_IN, r._DNS_HOST_TTL)
+        time.sleep(1)
+        record2 = r.DNSRecord('irrelevant', r._TYPE_SRV, r._CLASS_IN, r._DNS_HOST_TTL)
+
+        assert record.created != record2.created
+        assert record._expiration_time != record2._expiration_time
+        assert record._stale_time != record2._stale_time
```
Comment (jstasiak) [inline zeroconf/test.py]: I think accessing private attributes is not the way to do this. If there's some internal change it should be testable using the public API of the `DNSRecord` class (an API that uses `_expiration_time` and `_stale_time` internally that is). The candidates for this are:

* `get_remaining_ttl()` (uses `_expiration_time`)
* `is_expired()` (`_expiration_time`)
* `is_stale()` (`_stale_time`)

File: zeroconf/test.py [hunk 5]
```diff
@@ -1353,3 +1373,7 @@ def send(out, addr=r._MDNS_ADDR, port=r._MDNS_PORT):
 
     # unregister
     zc.unregister_service(info)
+
+
+if __name__ == '__main__':
```
Comment (jstasiak) [inline zeroconf/test.py]: Sorry, but this is also out of scope of this pull request.

File: zeroconf/test.py (hunk 2 above)
Comment (mattsaxon) [reply zeroconf/test.py]: Thanks for the explanation, will do.

File: zeroconf/test.py (hunk 3 above)
Comment (mattsaxon) [reply zeroconf/test.py]: Understood

File: zeroconf/test.py (hunk 4 above)
Comment (mattsaxon) [reply zeroconf/test.py]: Updated in next PR

File: zeroconf/test.py (hunk 5 above)
Comment (mattsaxon) [reply zeroconf/test.py]: OK, will remove, will you accept this in another PR?

Comment (jstasiak) [reply zeroconf/test.py]: I don't think it'd be useful – even now I don't think we can fully run the test suite without `nosetests` (not every test is in a test method in a subclass of `unittest.TestCase`) and I plan to change it further in the future (less subclassing of `TestCase`).

**Comment by jstasiak** [comment]
Thanks, that's a nice one!

**Comment by jstasiak** [comment]
Released as part of 0.24.4 just now.
